airbyte-integrations/connectors/source-relational-db/src/main/java/io/airbyte/integrations/source/relationaldb/AbstractDbSource.java

```diff
@@ -509,4 +512,24 @@ private Database createDatabaseInternal(final JsonNode sourceConfig) throws Exce
     return database;
   }
 
+  /**
+   * Deserializes the state represented as JSON into an object representation.
+   *
+   * @param stateJson The state as JSON.
+   * @return The deserialized object representation of the state.
+   */
+  protected List<AirbyteStateMessage> deserializeState(final JsonNode stateJson) {
```

Comment:
This is what we were discussing today. In an ideal world it would be great if this could live on the other side of the `Source` interface. Until we have versioning it has to be here.